web/packages/teleport/src/Roles/RoleEditor/RoleEditor.test.tsx

```diff
@@ -204,24 +204,89 @@ test('no double conversions when clicking already active tabs', async () => {
   );
 });
 
-test('canceling standard editor', async () => {
-  const onCancel = jest.fn();
-  render(<TestRoleEditor onCancel={onCancel} />);
-  await user.click(screen.getByRole('button', { name: 'Cancel' }));
-  expect(onCancel).toHaveBeenCalled();
-  expect(userEventService.captureUserEvent).toHaveBeenCalledWith({
-    event: CaptureEvent.CreateNewRoleCancelClickEvent,
+describe('closing the editor', () => {
+  function originalRole() {
+    const object = withDefaults({
+      kind: 'role',
+      metadata: {
+        name: 'new_role_name',
+      },
+    });
+    return { object, yaml: toFauxYaml(object) };
+  }
+
+  function setup(props: Partial<RoleEditorProps> = {}) {
+    const onCancel = jest.fn();
+    render(<TestRoleEditor onCancel={onCancel} {...props} />);
+    return onCancel;
+  }
+
+  async function closeWithConfirmation(onCancel: () => void) {
```

Comment:
I like those little helper functions, maybe I should be doing this more.

Reply:
Yeah, some of these tests would be very verbose otherwise.